table_foosball/table_foosball_01.rb

```diff
@@ -0,0 +1,141 @@
+require "securerandom"
```

Comment:
Overall comment about the usage of comments in code:

Most of the time a comment indicates that you should abstract something to well-named method. For example:

```ruby
class Animal
  def initialize(name, birthdate)
    @name = name
    @birthdate = birthdate 
  end
end

class Walrus < Animal
  def sexual_maturity
    7.years
  end

  def rut_months
   (1..4)
  end
end

class Herd
  def initialize(animals)
    @animals = animals
  end

  def in_rut
    # if we're not in rut, then no animals in the herd are
    return [] unless animal.first.rut_months.include?(Time.now.month)

    # select all the animals that are older than the sexual maturity age
    animals.select { |animal| 
       Time.now - animal.birthdate > animal.sexual_maturity 
    }
  end
end
```

While this code is not terrible, the `#in_rut` method is more complicated than it needs to be. By moving things around to the right place of ownership, we can make the code more clear, so `#in_rut` now looks like:

```ruby
 def in_rut
    return [] unless animals.first.in_rut?
    animals.select { |animal| animal.sexually_mature? }
  end
```

In this case more responsibility is moved to the class that inherits from `Animal` to make the code much more clear.